src/main/java/org/junitpioneer/jupiter/ReportEntryExtension.java

```diff
@@ -60,16 +96,35 @@ public void testAborted(ExtensionContext context, Throwable cause) {
 
 	@Override
 	public void testFailed(ExtensionContext context, Throwable cause) {
-		publishOnConditions(context, ALWAYS, ON_FAILURE);
+		if (!(cause instanceof ExtensionConfigurationException)) {
+			publishOnConditions(context, ALWAYS, ON_FAILURE);
+		}
 	}
 
 	private void publishOnConditions(ExtensionContext context, ReportEntry.PublishCondition... conditions) {
 		findAnnotations(context)
 				.filter(entry -> Arrays.asList(conditions).contains(entry.when()))
-				// we filter for empty keys/values because this is called if the test failed -
-				// even if it's due to bad extension configuration (but we don't publish for those)
-				.filter(entry -> !entry.key().isEmpty() && !entry.value().isEmpty())
-				.forEach(entry -> context.publishReportEntry(entry.key(), entry.value()));
+				.forEach(entry -> context.publishReportEntry(entry.key(), parseVariables(entry.value(), context)));
+	}
+
+	private String parseVariables(String value, ExtensionContext context) {
+		if (!value.matches(".*\\{.*}.*"))
```

Comment:
Shouldn't there be a start of line and end of line symbol in the regEx?

Reply:
I don't understand the regex. The `\\` means you check for `\` (first backslash is to escape second baskslash), but why even check for a baskslash? Then inside the curly braces you check for any character which may or may not be there (there can be no character between the curly braces). But I think (look below in line 117) you search for numeric (interger) values. So why you don't check for them inside the regex?

Reply:
- We don't really care where we find the regex inside the `String` so there is no need for [boundary matchers](https://docs.oracle.com/javase/tutorial/essential/regex/bounds.html).
- `{` and `}` are 'special' characters in regexes (quantifiers?). For example `a{5}` means _exactly_ five `a` characters. Because of this, the opening `{` needs to be "escaped" in the regex. Normally this would be done with `\{` however, in Java, `\` is an escape character already (e.g.: `\n` is newline). So, we have to put `\\` in the regex - first, to escape from the String, then to escape from the regex. Worse than Alcatraz, really.
- This is an early return, checking for a negative - if there are no curly braces in the value, we return immediately because there are no variables in the `@ReportEntry` value. If there are, we try to parse them and throw an exception if they cannot be parsed. An argument could be made that empty curly braces, i.e.: `{}` should not throw an exception - if you think that should be changed, let me know.

Reply:
I should stop reviewing, of course `\\{` is the regex and not `\{` 🤦